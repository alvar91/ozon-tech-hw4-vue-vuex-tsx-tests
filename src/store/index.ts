import Vue from "vue";
import Vuex from 'vuex';
import { createVuexStore } from "vuex-simple";

// @ts-ignore
import { MainModule } from "./modules/main/main.ts";

Vue.use(Vuex);

const store = createVuexStore(new MainModule());

export default store;
