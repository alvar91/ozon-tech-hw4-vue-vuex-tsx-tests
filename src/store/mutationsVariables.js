export const FETCH_USERS_ACTION = "FETCH_USERS_ACTION";
export const FETCH_USERS = "FETCH_USERS";
export const SET_LOADING = "SET_LOADING";
export const SET_CURRENT_USER = "SET_CURRENT_USER";
export const SET_FILTER_KEYWORD = "SET_FILTER_KEYWORD";
export const SET_FORM_INPUT = "SET_FORM_INPUT";
export const SET_FORM_CHANGE = "SET_FORM_CHANGE";
export const SET_FORM_RADIO = "SET_FORM_RADIO";
