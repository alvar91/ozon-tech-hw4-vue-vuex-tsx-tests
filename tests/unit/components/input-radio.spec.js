import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";

import InputRadio from "@/components/input-radio.vue";

import {
  mockUserId,
  mockUser,
} from "../../mock/mockUser";

import {
  FETCH_USERS_ACTION,
  SET_CURRENT_USER,
} from "@/store/mutationsVariables";

import store from "@/store/index";

const localVue = createLocalVue();
localVue.use(Vuex);

const { gender } = mockUser;
describe("InputRadio.vue", () => {
  let wrapper;

  beforeEach(async () => {
    await store.dispatch(FETCH_USERS_ACTION);
    store.commit(SET_CURRENT_USER, mockUserId);
  });

  it("should correctly renders markup with default props", () => {
    wrapper = mount(InputRadio, {
      localVue,
      propsData: {
        title: "Пол",
        name: "gender",
        value: { gender },
      },
      store,
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should correctly input data and change the store", () => {
    wrapper = mount(InputRadio, {
      localVue,
      propsData: {
        title: "Пол",
        name: "gender",
        value: { gender },
      },
      store,
    });

    const prevGender = store.getters.currentUser.gender;

    const inputs = wrapper.findAll(".card__input-radio");
    const maleInput = inputs.at(0);
    const femaleInput = inputs.at(1);

    if (store.getters.currentUser.gender === "male") {
      femaleInput.trigger("change");
    } else {
      maleInput.trigger("change");
    }

    expect(store.getters.currentUser.gender).not.toBe(prevGender);
  });

  afterEach(() => {
    wrapper.destroy();
  });
});
