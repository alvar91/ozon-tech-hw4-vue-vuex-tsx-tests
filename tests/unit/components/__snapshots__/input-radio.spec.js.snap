// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRadio.vue should correctly renders markup with default props 1`] = `<span class="card__field"><span class="card__title">Пол</span><span class="card__label-container"><label tabindex="0" class="card__radio">Мужской<input type="radio" name="gender" class="card__input-radio" value="Мужской"><span class="card__checkmark"></span></label><label tabindex="0" class="card__radio">Женский<input type="radio" name="gender" class="card__input-radio" value="Женский"><span class="card__checkmark"></span></label></span></span>`;
